File: src/Modules/Notification/WorkBase.Modules.Notification.Domain/Entities/Notification.cs
using WorkBase.Shared.Domain;

namespace WorkBase.Modules.Notification.Domain.Entities;

public sealed class Notification : AuditableEntity<Guid>, ITenantScoped, IAuditable
{
    public Guid TenantId { get; private set; }
    public Guid RecipientUserId { get; private set; }
    public string Title { get; private set; } = string.Empty;
    public string Body { get; private set; } = string.Empty;
    public string Category { get; private set; } = string.Empty;
    public bool IsRead { get; private set; }
    public DateTime? ReadAt { get; private set; }
    public string? ReferenceType { get; private set; }
    public Guid? ReferenceId { get; private set; }

    private Notification() { }

    public static Notification Create(
        Guid tenantId,
        Guid recipientUserId,
        string title,
        string body,
        string category,
        string? referenceType = null,
        Guid? referenceId = null)
    {
        return new Notification
        {
            TenantId = tenantId,
            RecipientUserId = recipientUserId,
            Title = title,
            Body = body,
            Category = category,
            ReferenceType = referenceType,
            ReferenceId = referenceId
        };
    }

    public void MarkAsRead()
    {
        if (IsRead) return;
        IsRead = true;
        ReadAt = DateTime.UtcNow;
    }
}

File: frontend/src/api/types/notification.ts
export interface NotificationDto {
  id: string;
  title: string;
  body: string;
  category: string;
  isRead: boolean;
  createdAt: string;
  readAt: string | null;
  referenceType: string | null;
  referenceId: string | null;
}

export interface NotificationTemplateDto {
  id: string;
  code: string;
  name: string;
  titleTemplate: string;
  bodyTemplate: string;
  category: string;
  isActive: boolean;
}

export interface CreateNotificationTemplateRequest {
  code: string;
  name: string;
  titleTemplate: string;
  bodyTemplate: string;
  category: string;
}

export interface UpdateNotificationTemplateRequest {
  name: string;
  titleTemplate: string;
  bodyTemplate: string;
  category: string;
}
